moshi/src/main/java/com/squareup/moshi/internal/Util.kt

```diff
@@ -499,6 +499,19 @@ public fun <T> Class<T>.boxIfPrimitive(): Class<T> {
   return wrapped ?: this
 }
 
+internal inline fun <T : Any> requireNull(value: T?, lazyMessage: (T) -> Any) {
+  contract {
+    returns() implies (value == null)
+  }
```

Comment:
This isn't really necessary since you're using the value in the message creation instead

Reply:
https://github.com/square/moshi/pull/1467/commits/fc4f88186616bc7f64ba5fbdcb9b472e4e836a34